### ISSUE TITLE
[Bug]: will-navigate doesn't fire when navigating to about: URLs

### ISSUE THREAD

**Issue by pushkin-** [electron/electron#21136]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for an issue that matches the one I want to file, without success.

### Issue Details

* **Electron Version:** 5.0.7
* **Operating System:** Windows 10

### Expected Behavior
Based on the documentation, I expect that navigating to `"about:blank"` will trigger the webcontent's `will-navigate` event.

### Actual Behavior
`will-navigate` does not get triggered.

The [docs](https://electronjs.org/docs/api/web-contents#event-will-navigate) make no mention of `"about:blank"` being an exception, and so I expect that `will-navigate` should get emitted when navigating to it.

Either this is a documentation error, or it's a bug with `will-navigate`.

### To Reproduce
```
const { app, BrowserWindow, BrowserView } = require('electron')

async function onReady() {
	const mainWindow = new BrowserWindow();
	const view = new BrowserView();
	view.webContents.addListener("will-navigate", (e, url) => {
		console.log("navigating to: " + url);
	});
	mainWindow.setBrowserView(view);
	await view.webContents.loadURL("https://google.com");
	view.webContents.openDevTools();
}

app.on('ready', onReady)
```

After `npm start`ing, in the devtools, run `location = "https://google.com"`. Notice that we log it.

Then run `location = "about:blank"`. Notice, that there's no log.

**Comment by pushkin-** [comment]
this is also an issue with Electron 7.1.2

**Comment by pushkin-** [comment]
Still a problem in Electron 10.1.2

**Comment by pimterry** [comment]
Still a problem in Electron v11.3.0, and a fairly annoying one: I include links from external content in my app, and intercept them using `will-navigate` and `new-window` to allow HTTP links only and to redirect them to the system browser. It seems that `about:blank` ignores all that, and somehow some `about:blank` links are occasionally sneaking into some of the content I render.

If anybody clicks such a link, instead of either being blocked or opening in the system browser (both pretty reasonable) it blanks the entire app. As far as I can tell there's no other event or workaround for this available at present, so there's nothing I can do to protect against this at the Electron level.

**Comment by pushkin-** [comment]
@pimterry Can you use [`onBeforeRequest`](https://www.electronjs.org/docs/api/web-request#webrequestonbeforerequestfilter-listener) to cancel the request to about:blank?

**Comment by pimterry** [comment]
@pushkin- no, that doesn't work either, it doesn't fire (presumably because there's no actual HTTP request involved)

**Comment by mckravchyk** [comment]
Note that it's not just `about:blank`, but also `about:`, `about:anything`

**Comment by rzhao271** [comment]
The issue still occurs in 16.0.4.
gist: https://gist.github.com/rzhao271/db5a33e5b332bc5f292f049be6c404a1

**Comment by CodeSmith32** [comment]
Bump:
This is still an issue as of Electron v20.0.0.

**Comment by CodeSmith32** [comment]
Bump:
This is still an issue as of Electron v22.0.0.

**Comment by pushkin-** [comment]
bump

**Comment by pushkin-** [comment]
bump

**Comment by pushkin-** [comment]
bump

**Comment by pushkin-** [comment]
repros in v42

**Comment by pushkin-** [comment]
bump
